src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx

```diff
@@ -492,11 +494,12 @@ function ScatterplotViz(props: VisualizationProps) {
                 ? true
                 : false
               : allData.facets
+                  .filter((facet) => facet.data != null)
                   .map(
                     ({ data }) =>
-                      data.series[index]?.y != null &&
-                      data.series[index].y.length > 0 &&
-                      data.series[index].y[0] !== null
+                      data?.series[index]?.y != null &&
```

Comment:
I already filtered for facet.data not null.  Not sure why the `?` is still needed.

Reply:
There is a long backstory to this, but you can you a type guard annotation in the `filter` callback to prevent this. It would basically look like this:

```typescript
.map(facet => facet.data)
.filter((data): data is D => data != null)
```

where `D` is the non-null type of `facet.data`.